src/main/java/com/aimmac23/hub/videostorage/CloudS3StoredVideoInfoContext.java

```diff
@@ -0,0 +1,53 @@
+package com.aimmac23.hub.videostorage;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Context information for AWS S3 stored videos
+ */
+public class CloudS3StoredVideoInfoContext implements StoredVideoInfoContext {
+	private final LocationAwareS3Object video;
+
+	private static final Logger log = Logger.getLogger(CloudS3StoredVideoInfoContext.class.getName());
+
+	CloudS3StoredVideoInfoContext(LocationAwareS3Object videoObject) {
+		this.video = videoObject;
+	}
+
+	@Override
+	public boolean isVideoFound() {
+		try {
+			return (video != null && video.getS3Object().getObjectContent().available() > 0);
+		} catch (IOException e) {
+			return false;
+		}
+	}
+
+	@Override
+	public Long getContentLengthIfKnown() {
+		try {
+			return video != null ? (long) video.getS3Object().getObjectContent().available() : 0L;
+		} catch (IOException e) {
+			return 0L;
```

Comment:
Returning NULL in case of error or missing file is more appropriate (see interface javadoc)